build.gradle

```diff
@@ -101,29 +110,37 @@ task sourcesJar(type: Jar) {
     from sourceSets.main.allSource
 }
 
-publishing {
-    publications {
-        java(MavenPublication) {
-            artifact jar
+if (!System.env.RELEASE) {
+    build.dependsOn 'publishToMavenLocal'
```

Comment:
every time you'll run `build` it'll publish to `mavenLocal()` which is useless. You better have to make a dependency in the parent build to `publishToMavenLocal`